File: examples/matlab/kmeans/basic/simple_from_matrix.m
function simple_from_matrix()
    addpath(fileparts(fileparts(fileparts(mfilename('fullpath')))));
    check_compile('kmeans')
    
    % This example shows how to cluster a matlab matrix
    % Only sparse matrices with double values are supported!
    X = sprand(1000, 200, 1/10);
    
    k = 10;
    % If the dataset has M samples, then IDX is an Mx1 matrix assigning each sample the closest cluster index.   
    [ IDX ] = fcl_kmeans(X, k);
    
    % Second example:
    % Additionally to IDX, the cluster centers matrix C is outputted.
    [ IDX, C ] = fcl_kmeans(X, k);
    
    % Third example:
    % Additionally to IDX and C, sumd the within-cluster sums of point-to-centroid distances in the 1-by-k vector is returned 
    [ IDX, C, sumd] = fcl_kmeans(X, k);
end

File: examples/python/kmeans/basic/simple_from_matrix.py
from __future__ import print_function
from fcl import kmeans
import numpy as np
from scipy import sparse

if __name__ == "__main__":
    # Create random dataset
    np.random.seed(1) # set a random seed to always generate the same matrix X
    X = np.random.rand(100, 100)
    
    # Make matrix sparse by just removing values below 0.5
    X[X < 0.5] = 0
    sparse_X = sparse.csr_matrix(X)
    
    # this example shows how to cluster a numpy matrix.
    km = kmeans.KMeans(no_clusters=10, seed = 0)
    km.fit(sparse_X)
    
    # If the dataset has M samples, then idx is an Mx1 array assigning each sample the closest cluster index.   
    idx = km.predict(sparse_X)
    C = km.get_cluster_centers()   
    
    no_clusters, dim = C.shape
    
    # Get matrices of samples which are in the same cluster
    matrices = {}
    
    for i in range(no_clusters):
        matrices[i] = sparse_X[idx == i]
        print("Cluster %d has shape"%i, matrices[i].shape)
